csrc/ec_utils.cpp

```diff
@@ -155,9 +158,101 @@ JNIEXPORT jint JNICALL Java_com_amazon_corretto_crypto_provider_EcUtils_curveNam
         }
         orderBN.toJavaArray(env, orderArr);
 
+        // Get the decoded (string) curve OID
+        jni_borrow oidBorrow = jni_borrow(env, java_buffer::from_array(env, oid), "curveNameToInfo");
+        if (!OBJ_obj2txt((char*) oidBorrow.data(), (int) oidBorrow.len(), OBJ_nid2obj(nid), /*always_return_oid*/1)) {
+            throw_openssl("Unable to get decoded curve OID");
+        }
+        oidBorrow.release();
```

Comment:
yes, while `release` is [called by the dtor](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/csrc/buffer.h#L349) when a `jni_borrow` goes out of scope, we need to explicitly call it here before taking out the next borrow to avoid a segfault. if i don't manually release the first `jni_borrow`, i get this:

```
Attempted to use JNI environment while buffer locks were outstanding:
        0x7f85ddeb9df0: "curveNameToInfo"
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f85dd71e4f0, pid=27432, tid=27567
#
# JRE version: OpenJDK Runtime Environment Corretto-11.0.18.10.1 (11.0.18+10) (build 11.0.18+10-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-11.0.18.10.1 (11.0.18+10-LTS, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libamazonCorrettoCryptoProvider.so+0x1b4f0]  Java_com_amazon_corretto_crypto_provider_EcUtils_curveNameToInfo+0x4f0
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /local/home/childw/workplace/github/corretto/amazon-corretto-crypto-provider/build/cmake/hs_err_pid27432.log
End trace (1 locks)
```

why do we get a segfault if we fail to manually release the first `jni_borrow` before taking out the second? i'm not 100% sure, as I found the manual release workaround and kept going. but digging into it a little, i suspect that it has to do with how we manage our own locks on top of the JNI functions or with [our usage of GetPrimitiveArrayCritical](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/csrc/buffer.h#L333). to my mind, it still shouldn't be an issue as the arrays don't overlap (why would we need to lock?), and the C++ compiler should call the dtors in the correct order when they go out of scope (LIFO -- first `encodedBorrow`, then `oidBorrow`). if we do need to take out multiple `jni_borrow`s at a time in the same scope, we'll need to investigate this further.

i'm using `jni_borrow` here for performance reasons -- we want to write directly to the java heap if we can to avoid copies.

the `release` call on 175 is unnecessary; i included it for consistency.